src/mlpack/tests/hdbscan_test.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file hdbscan_test.cpp
+ *
+ * Test the HDBSCAN implementation.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/hdbscan/hdbscan.hpp>
+
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::hdbscan;
+
+
+BOOST_AUTO_TEST_SUITE(HDBSCANTest);
+
+BOOST_AUTO_TEST_CASE(singleClusterTest)
+{
+  arma::mat points;
+  if (!data::Load("circle.csv", points))
+    BOOST_FAIL("Cannot load test dataset circle.csv!");
+
+  HDBSCAN<> d1(5, true);
+  arma::Row<size_t> assignments;
+  d1.Cluster(points, assignments);
+
+  //Some points in the circle will get classified as noise
+  for (size_t i = 0; i < assignments.n_cols; i++)
+  {
+    BOOST_REQUIRE((assignments(i)==0) || (assignments(i)==SIZE_MAX));
+  }
+}
+
+BOOST_AUTO_TEST_CASE(noiseTest)
+{
+  arma::mat points;
+  if (!data::Load("circle.csv", points))
+    BOOST_FAIL("Cannot load test dataset circle.csv!");
```

Comment:
Is the circle dataset just points on the 2-d circle?  If so can you generate this data instead?